### PR TITLE
Move Dev Guide to collection

### PR DESCRIPTION
**Pull request by mariolenz**
Depends-On: https://github.com/ansible-collections/vmware.vmware_rest/pull/331

##### SUMMARY
Collection-specific docs must move to their collections.

##### ISSUE TYPE
- Docs Pull Request

##### COMPONENT NAME
docsite

##### ADDITIONAL INFORMATION
see [here](https://github.com/ansible-community/community-topics/issues/91#issuecomment-1117439732)

cc @samccann

### CODE REVIEW

**Comment by mariolenz** [comment]
Any idea why the CI fails with `./docs/docsite/rst/dev_guide.rst:56:0: (ERROR/3) Unknown interpreted text role "file"`? There wasn't a problem in `ansible/ansible`.

**Comment by Akasurde** [comment]
> Any idea why the CI fails with `./docs/docsite/rst/dev_guide.rst:56:0: (ERROR/3) Unknown interpreted text role "file"`? There wasn't a problem in `ansible/ansible`.

I think this line is causing error - https://github.com/ansible-collections/vmware.vmware_rest/pull/330/files#diff-0b510edc985da022dbbc809c141a2cce98c926d0f44c61533792248231bffdbbR56

Not sure why.

File: docs/docsite/rst/dev_guide.rst [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+      :local:
+
+Contribution process
+====================
+
+The modules of the vmware_rest collection are autogenerated by another tool called `vmware_rest_code_generator <https://github.com/ansible-collections/vmware_rest_code_generator>`.
+
+If you would like to contribute a change, we would appreciate if you:
+
+- submit a Github Pull Request (PR) against the vmware_rest_code_generator project
```
Comment (Akasurde) [inline docs/docsite/rst/dev_guide.rst:19]:
```suggestion
- submit a GitHub Pull Request (PR) against the vmware_rest_code_generator project
```

**Comment by mariolenz** [comment]
@Akasurde I've removed the offending `:file:`, although I'm not really sure what this implies.

But these failures don't look related to my changes:

```
2022-05-06 08:42:05.407596 | controller | ERROR: Found 24 validate-modules issue(s) which need to be resolved:
2022-05-06 08:42:05.407702 | controller | ERROR: plugins/lookup/cluster_moid.py:0:0: invalid-documentation: DOCUMENTATION.author: Invalid author for dictionary value @ data['author']. Got ['Alina Buzachis <@alinabuzachis>']
2022-05-06 08:42:05.407775 | controller | ERROR: plugins/lookup/cluster_moid.py:0:0: invalid-documentation: DOCUMENTATION.lookup: extra keys not allowed @ data['lookup']. Got 'cluster_moid'
2022-05-06 08:42:05.407828 | controller | ERROR: plugins/lookup/cluster_moid.py:0:0: invalid-documentation: DOCUMENTATION.name: required key not provided @ data['name']. Got None
2022-05-06 08:42:05.407886 | controller | ERROR: plugins/lookup/datacenter_moid.py:0:0: invalid-documentation: DOCUMENTATION.author: Invalid author for dictionary value @ data['author']. Got ['Alina Buzachis <@alinabuzachis>']
2022-05-06 08:42:05.407935 | controller | ERROR: plugins/lookup/datacenter_moid.py:0:0: invalid-documentation: DOCUMENTATION.lookup: extra keys not allowed @ data['lookup']. Got 'datacenter_moid'
2022-05-06 08:42:05.407985 | controller | ERROR: plugins/lookup/datacenter_moid.py:0:0: invalid-documentation: DOCUMENTATION.name: required key not provided @ data['name']. Got None
2022-05-06 08:42:05.408039 | controller | ERROR: plugins/lookup/datastore_moid.py:0:0: invalid-documentation: DOCUMENTATION.author: Invalid author for dictionary value @ data['author']. Got ['Alina Buzachis <@alinabuzachis>']
2022-05-06 08:42:05.408087 | controller | ERROR: plugins/lookup/datastore_moid.py:0:0: invalid-documentation: DOCUMENTATION.lookup: extra keys not allowed @ data['lookup']. Got 'datastore_moid'
2022-05-06 08:42:05.408169 | controller | ERROR: plugins/lookup/datastore_moid.py:0:0: invalid-documentation: DOCUMENTATION.name: required key not provided @ data['name']. Got None
2022-05-06 08:42:05.408224 | controller | ERROR: plugins/lookup/folder_moid.py:0:0: invalid-documentation: DOCUMENTATION.author: Invalid author for dictionary value @ data['author']. Got ['Alina Buzachis <@alinabuzachis>']
2022-05-06 08:42:05.408273 | controller | ERROR: plugins/lookup/folder_moid.py:0:0: invalid-documentation: DOCUMENTATION.lookup: extra keys not allowed @ data['lookup']. Got 'folder_moid'
2022-05-06 08:42:05.408323 | controller | ERROR: plugins/lookup/folder_moid.py:0:0: invalid-documentation: DOCUMENTATION.name: required key not provided @ data['name']. Got None
2022-05-06 08:42:05.408378 | controller | ERROR: plugins/lookup/host_moid.py:0:0: invalid-documentation: DOCUMENTATION.author: Invalid author for dictionary value @ data['author']. Got ['Alina Buzachis <@alinabuzachis>']
2022-05-06 08:42:05.408426 | controller | ERROR: plugins/lookup/host_moid.py:0:0: invalid-documentation: DOCUMENTATION.lookup: extra keys not allowed @ data['lookup']. Got 'host_moid'
2022-05-06 08:42:05.408475 | controller | ERROR: plugins/lookup/host_moid.py:0:0: invalid-documentation: DOCUMENTATION.name: required key not provided @ data['name']. Got None
2022-05-06 08:42:05.408529 | controller | ERROR: plugins/lookup/network_moid.py:0:0: invalid-documentation: DOCUMENTATION.author: Invalid author for dictionary value @ data['author']. Got ['Alina Buzachis <@alinabuzachis>']
2022-05-06 08:42:05.408577 | controller | ERROR: plugins/lookup/network_moid.py:0:0: invalid-documentation: DOCUMENTATION.lookup: extra keys not allowed @ data['lookup']. Got 'network_moid'
2022-05-06 08:42:05.408626 | controller | ERROR: plugins/lookup/network_moid.py:0:0: invalid-documentation: DOCUMENTATION.name: required key not provided @ data['name']. Got None
2022-05-06 08:42:05.408679 | controller | ERROR: plugins/lookup/resource_pool_moid.py:0:0: invalid-documentation: DOCUMENTATION.author: Invalid author for dictionary value @ data['author']. Got ['Alina Buzachis <@alinabuzachis>']
2022-05-06 08:42:05.408728 | controller | ERROR: plugins/lookup/resource_pool_moid.py:0:0: invalid-documentation: DOCUMENTATION.lookup: extra keys not allowed @ data['lookup']. Got 'resource_pool_moid'
2022-05-06 08:42:05.408777 | controller | ERROR: plugins/lookup/resource_pool_moid.py:0:0: invalid-documentation: DOCUMENTATION.name: required key not provided @ data['name']. Got None
2022-05-06 08:42:05.408830 | controller | ERROR: plugins/lookup/vm_moid.py:0:0: invalid-documentation: DOCUMENTATION.author: Invalid author for dictionary value @ data['author']. Got ['Alina Buzachis <@alinabuzachis>']
2022-05-06 08:42:05.408878 | controller | ERROR: plugins/lookup/vm_moid.py:0:0: invalid-documentation: DOCUMENTATION.lookup: extra keys not allowed @ data['lookup']. Got 'vm_moid'
2022-05-06 08:42:05.408931 | controller | ERROR: plugins/lookup/vm_moid.py:0:0: invalid-documentation: DOCUMENTATION.name: required key not provided @ data['name']. Got None
2022-05-06 08:42:05.409219 | controller | See documentation for help: https://docs.ansible.com/ansible-core/devel/dev_guide/testing/sanity/validate-modules.html
```

**Comment by Akasurde** [comment]
Yes, this is related to docs in the lookup plugin `plugins/lookup/` 

cc @alinabuzachis

**Comment by mariolenz** [comment]
> Yes, this is related to docs in the lookup plugin `plugins/lookup/`

Yes, you're right! I think sanity testing lookup plugins is quite new, so that's why it didn't fail previously. I think I know what the problem is here, let me try something.

**Comment by alinabuzachis** [comment]
@mariolenz Oh, let me know if I can do something and give it a try! Thank you.

**Comment by mariolenz** [comment]
@Akasurde @alinabuzachis Let's see if #331 helps.

**Comment by mariolenz** [comment]
recheck

**Comment by mariolenz** [comment]
@Akasurde Could you please have a look at this PR again?

**Pull request merged**